src/csharp/Grpc.Core/Internal/NativeExtension.cs

```diff
@@ -78,29 +80,80 @@ public NativeMethods NativeMethods
         }
 
         /// <summary>
-        /// Detects which configuration of native extension to load and load it.
+        /// Detects which configuration of native extension to load and explicitly loads the dynamic library.
+        /// The explicit load makes sure that we can detect any loading problems early on.
         /// </summary>
-        private static NativeMethods LoadNativeMethodsLegacyNetFramework()
+        private static NativeMethods LoadNativeMethodsUsingExplicitLoad()
         {
-            // TODO: allow customizing path to native extension (possibly through exposing a GrpcEnvironment property).
-            // See https://github.com/grpc/grpc/pull/7303 for one option.
+            // NOTE: a side effect of searching the native extension's library file relatively to the assembly location is that when Grpc.Core assembly
+            // is loaded via reflection from a different app's context, the native extension is still loaded correctly
+            // (while if we used [DllImport], the native extension won't be on the other app's search path for shared libraries).
             var assemblyDirectory = GetAssemblyDirectory();
 
             // With "classic" VS projects, the native libraries get copied using a .targets rule to the build output folder
             // alongside the compiled assembly.
+            // With dotnet SDK projects targeting net45 framework, the native libraries (just the required ones)
+            // are similarly copied to the built output folder, through the magic of Microsoft.NETCore.Platforms.
             var classicPath = Path.Combine(assemblyDirectory, GetNativeLibraryFilename());
 
+            // With dotnet SDK project targeting netcoreappX.Y, projects will use Grpc.Core assembly directly in the location where it got restored
```

Comment:
this just adds back code that was removed in https://github.com/grpc/grpc/pull/24744/files#diff-2c65a0fc0ff30ef55102682b95465c44e758e53921b0f36be1b427f1feb12e18L95